src/aiperf/gpu_telemetry/metrics_config.py

```diff
@@ -0,0 +1,251 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""GPU telemetry metrics configuration utilities."""
+
+import re
+from pathlib import Path
+
+from aiperf.common.enums.metric_enums import (
+    EnergyMetricUnit,
+    FrequencyMetricUnit,
+    GenericMetricUnit,
+    MetricSizeUnit,
+    MetricTimeUnit,
+    MetricUnitT,
+    PowerMetricUnit,
+    TemperatureMetricUnit,
+)
+from aiperf.common.mixins import AIPerfLoggerMixin
+from aiperf.gpu_telemetry import constants
+
+
+def _title_case_metric_name(name: str) -> str:
+    """
+    Convert metric name to proper title case with special handling for acronyms.
+
+    Handles special cases like GPU, XID, SM, etc. to remain fully capitalized.
+
+    Args:
+        name: The metric name to convert
+
+    Returns:
+        Title-cased metric name with proper acronym capitalization
+
+    Example:
+        >>> _title_case_metric_name("gpu power usage")
+        "GPU Power Usage"
+        >>> _title_case_metric_name("xid errors")
+        "XID Errors"
+        >>> _title_case_metric_name("sm clock frequency")
+        "SM Clock Frequency"
+    """
+    # List of acronyms that should be fully capitalized
+    acronyms = {
+        "gpu",
+        "xid",
+        "sm",
+        "nvlink",
+        "pci",
+        "pcie",
+        "cpu",
+        "ram",
+        "vram",
+        "ecc",
+    }
```

Comment:
i hate how the formatter does this one per-line crap.

not a big deal, but maybe you can try like this:

```python
    acronyms = { 
        "gpu", "xid", "sm", "nvlink", "pci", 
        "pcie", "cpu", "ram", "vram", "ecc",
    }  # fmt: skip
```